state/protocol/badger/snapshot.go

```diff
@@ -72,6 +72,10 @@ func (s *Snapshot) Head() (*flow.Header, error) {
 	return head, err
 }
 
+func (s *Snapshot) BlockID() (flow.Identifier, error) {
```

Comment:
[Some modules](https://github.com/onflow/flow-go/blob/leo/add-block-id-to-state/engine/execution/ingestion/loader/unexecuted_loader.go#L160) already depend on protocol state, however, they still import `storage.Headers`, because they need to find blockID by height using the headers.BlockIDByHeight method. 

With the `BlockID` method added to snapshot along with the changes in the AtHeight method as below, we could use `state.AtHeight(height).BlockID()` to replace `headers.BlockIDByHeight`, which has been [used a lot](https://github.com/onflow/flow-go/pull/5175/files)